src/cc/frontends/p4/compiler/ebpfScalarType.py

```diff
@@ -0,0 +1,78 @@
+from p4_hlir.hlir import P4_AUTO_WIDTH
+from ebpfType import *
+from compilationException import *
+from programSerializer import ProgramSerializer
+
+
+class EbpfScalarType(EbpfType):
+    __doc__ = "Represents a scalar type"
+    def __init__(self, parent, widthInBits, isSigned, config):
+        EbpfType.__init__(self, None)
```

Comment:
Here I would request using super(EbpfType, self).__init__(None) as per http://stackoverflow.com/questions/576169/understanding-python-super-with-init-methods. The same applies elsewhere the parent constructor is called. This change necessitates converting to inheriting from object as well.